bonfire/bonfire.py

```diff
@@ -1016,8 +1078,16 @@ def _cmd_config_deploy(
     if not has_clowder():
         _error("cluster does not have clowder operator installed")
 
+    using_current = False
+    if reserve:
```

Comment:
```suggestion
    if conf.BONFIRE_BOT or reserve:
```

I think we should also force bonfire to never run with 'using_current' mode if `conf.BONFIRE_BOT` is True -- that way our PR check behavior doesn't change.

Our CICD scripts run in 'bot' mode: https://github.com/RedHatInsights/bonfire/blob/master/cicd/bootstrap.sh#L21

This option is already used here for another purpose: https://github.com/RedHatInsights/bonfire/blob/d2dec39cca3e304efa69d72a9986344c0f2f45d7/bonfire/bonfire.py#L139

Reply:
I've added it to the else if block, where it make sense. That would disable using namespace from current context if non is given for bot mode.

Reply:
Taking a closer look at this ... it should be OK if you type `bonfire deploy` without `--namespace` in bot mode... in the "current mode of operation" that would go ahead and reserve an ns for you and I think we want to maintain that behavior for any pr check scripts out there that call `bonfire deploy` without calling `bonfire namespace reserve` first.

Reply:
Ok, double check me on this ... would this logic be right?:

If 'reserve' is true then we force reserve a namespace.

Otherwise, if there is no namespace explicitly set, we will try to use current namespace as long as we are not in 'bot' mode. I also tweaked this to call `current_namespace_or_error`. If we are in 'bot' mode, then `using_current` stays set to False and a new namespace is reserved just like how things have always worked.

```python
    using_current = False
    if reserve:
        namespace = None
    elif not namespace and not conf.BONFIRE_BOT:
        using_current = True
        namespace = current_namespace_or_error()

    ns, reserved_new_ns = _get_namespace(
        namespace, name, requester, duration, pool, timeout, local, force,
        using_current=using_current
    )
```

Reply:
To avoid use of current namespace for bot let's change the condition but leave the rest as is:

```
    using_current = False
    if reserve:
        namespace = None
    elif not namespace and not conf.BONFIRE_BOT:
        using_current = True
        namespace = get_current_namespace()
```

`namespace` is `None` if not defined within bot command, so no force is required, and then it should follow original procedure with new namespace reservation.

Erroring out on no current namespace would break the original procedure with reservation, so keeping the original with `get_current_namespace()`.